File: Sort first half in ascending and second half in descending/Sort_first_half_in_ascending_and_second_half_in_descending.cpp
class Solution {
  public:
    
    vector<int> customSort(vector<int>& arr) {
        int n = arr.size();
        sort(arr.begin(),arr.begin()+(n/2));
        sort(arr.begin()+(n/2),arr.end(),greater<int>());
        return arr;
    }
};

File: Sort first half in ascending and second half in descending/Sort_first_half_in_ascending_and_second_half_in_descending.java
class Solution {
    public ArrayList<Integer> customSort(int[] arr) {
        int n = arr.length;
        ArrayList<Integer> list = new ArrayList<>();
        for (int num : arr) {
            list.add(num);
        }
        
        Collections.sort(list.subList(0, n / 2));
        Collections.sort(list.subList(n / 2, n), Collections.reverseOrder());
        
        return list;
    }
}